components/style/values/specified/position.rs

```diff
@@ -486,3 +490,179 @@ impl From<GridAutoFlow> for u8 {
         result
     }
 }
+
+#[cfg_attr(feature = "gecko", derive(MallocSizeOf))]
+#[derive(Clone, Debug, PartialEq)]
+/// Single template area
+pub struct TemplateAreas {
+    /// `named area` containing for each single template area
+    pub areas: Box<[NamedArea]>,
+    /// The original CSS string value of each single template area
+    pub strings: Box<[Box<str>]>,
+    /// width of each single template area
+    pub width: u32,
+}
+
+impl TemplateAreas {
+    /// Transform `vector` of str into single `template area`
+    pub fn from_vec(strings: Vec<Box<str>>) -> Result<TemplateAreas, ()> {
```

Comment:
nit: no need to be `pub`.

Reply:
Thanks for the catching!

Reply:
I found it's used in `shorthand` mako so maybe we still need it as a `pub` method?

https://github.com/servo/servo/blob/master/components/style/properties/shorthand/position.mako.rs#L316

Reply:
Oh, indeed, nvm then :)